Sofa/framework/Core/src/sofa/core/behavior/Constraint.inl

```diff
@@ -69,6 +69,7 @@ void Constraint<DataTypes>::buildConstraintMatrix(const ConstraintParams* cParam
 {
     if (cParams)
     {
+        m_constraintIndex.setValue(cIndex);
```

Comment:
@EulalieCoevoet what do you think about setting the value here, at a higher level? All derived classes would then set the value.

Reply:
you're right, setting the id could be done outside of the function `buildConstraintMatrix`. That is why I suggest https://github.com/sofa-framework/sofa/pull/4786.

This way, the id can be set from the outside using the `setConstraintId` function, that you can find here:

https://github.com/sofa-framework/sofa/blob/05a148594b72b495457fb2f1644edaa1c155925f/Sofa/framework/Simulation/Core/src/sofa/simulation/mechanicalvisitor/MechanicalBuildConstraintMatrix.cpp#L33

I suggest to pause this PR, the time that https://github.com/sofa-framework/sofa/pull/4786 is merged. Can you review it?